extensions/simple_extension.go

```diff
@@ -57,6 +59,7 @@ type TypeVariation struct {
 
 type Argument interface {
 	toTypeString() string
+	ArgType() (types.Type, error)
```

Comment:
I don't get this method on this interface. What's the point? It seems like you've replaced type assertions with custom method that returns errors. How's that better than doing a type assertion on ValueArg and then having ValueArg have a utility method? 

As part of this, I suggest we use unexported marker methods on Argument so it is clear that people can't extend it. There are only three types of arguments: ValuesArg, TypeArg and EnumArg.

Reply:
Removed this method and added marker